lib/job.js

```diff
@@ -1,167 +1,148 @@
-                    url += '?';
-                    _.forOwn(options, function(value, option) {
-                        url += `${option}=${value}`;
-                    })
-                }
-        
-                return request.post(url, {});
-            });
+    stop(qs) {
+        return this.post(`/jobs/${this._jobId}/_stop`, { qs });
     }
 
-    function ex_id() {
-        return request.get(`/jobs/${job_id}/ex`)
-            .then(function(job_spec) {
-                return job_spec.ex_id;
-            });
+    pause(qs) {
+        return this.post(`/jobs/${this._jobId}/_pause`, { qs });
     }
 
-    function status() {
-        return request.get(`/jobs/${job_id}/ex`)
-            .then(function(job_spec) {
-                return job_spec._status;
-            });
+    resume(qs) {
+        return this.post(`/jobs/${this._jobId}/_resume`, { qs });
+    }
+
+    recover(qs) {
+        return this.ex().then(exId => this.post(`/ex/${exId}/_recover`, { qs }));
+    }
+
+    ex() {
+        return this.get(`/jobs/${this._jobId}/ex`)
+            .then(jobSpec => jobSpec.ex_id);
+    }
+
+    status() {
+        return this.get(`/jobs/${this._jobId}/ex`)
+            .then(jobSpec => jobSpec._status);
     }
 
-    function waitForStatus(target, timeout) {
-        if (!timeout) timeout = 1000;
-
-        return new Promise(function(resolve, reject) {
-            // Not all terminal states considered failure.
-            var terminal = {
-                completed: false,
-                failed: true,
-                rejected: true,
-                aborted: true
-            };
-            function wait() {
-                status()
-                    .then(function(result) {
-                        if (result === target) {
-                            resolve(result);
-                        }
-                        else {
-                            setTimeout(wait, timeout);
-                        }
-
-                        // These are terminal states for a job so if we're not explicitly
-                        // watching for these then we need to stop waiting as the job
-                        // status won't change further.
-                        if (terminal[result] !== undefined) {
-                            if (terminal[result]) {
-                                reject(`Job has status: "${result}" which is terminal so status: "${target}" is not possible. job_id: ${job_id}`)
-                            }
-                            else {
-                                resolve(result)
-                            }
-                        }
-                    })
-                    .catch(function(err) {
-                        reject(err);
-                    })
+    waitForStatus(target, intervalMs = 1000, timeoutMs = 0) {
+        // Not all terminal states considered failure.
+        const terminal = {
+            completed: false,
+            failed: true,
```

Comment:
the list of terminal status are: ['completed', 'stopped', 'rejected', 'failed', 'terminated'] and those should be used as keys in the terminal object. Also completed is a terminal status but its being used to check if its an error scenario so its a bit confusing the way its being called and checked. Maybe change the name of the variable isFatalStatus to better reflect an error based termination vs general job termination